trino-s3-proxy/src/test/java/io/trino/s3/proxy/server/testing/ConfiguredTestingS3ProxyServer.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server.testing;
+
+import com.google.inject.Inject;
+import io.trino.s3.proxy.server.credentials.Credentials;
+import io.trino.s3.proxy.server.remote.PathStyleRemoteS3Facade;
+import io.trino.s3.proxy.server.testing.TestingConstants.TestingCredentials;
+
+import java.util.Optional;
+
+public class ConfiguredTestingS3ProxyServer
+{
+    @Inject
+    public ConfiguredTestingS3ProxyServer(TestingRemoteS3Facade delegatingS3Facade, ManagedS3MockContainer s3MockContainer, TestingCredentialsController testingCredentialsController, @TestingCredentials Credentials credentials)
```

Comment:
This class exists just to initialize a few testing shims. We can use Guice more fully and get rid of this class. wdyt?

a) We add a constructor to `TestingCredentialsController`

```java

    @Inject
    public TestingCredentialsController(@TestingCredentials Credentials testingCredentials)
    {
        addCredentials(testingCredentials);
    }
```

b) Make `ContainerS3Facade` top level and public. Bind it in Guice in the extension.

c) Change `ContainerS3Facaed`'s constructor:

```java

    @Inject
    public ContainerS3Facade(ManagedS3MockContainer s3MockContainer, TestingRemoteS3Facade delegatingS3Facade)
    {
        super(s3MockContainer.getContainerHost().host(), false, Optional.of(s3MockContainer.getContainerHost().port()));
        this.containerHost = s3MockContainer.getContainerHost();
        delegatingS3Facade.setDelegate(this);
    }
```

Reply:
Thanks for the suggestion! I didn't really like my approach either, but I prefered it over getting access to the injector and manually setting this up in different places. I'm no Guice expert, so any feedback is much appreciated!

**For (a):**
I thought about this possibility briefly at first, but I thought it polluted the `TestingCredentialsController` (which we may want to provide as a part of `proxy-testing` later on for people to run this, similarly to what Trino does) with specific implementation logic for this repo's test suite.

We can:
- Create a class that inherits from `TestingCredentialsController` and works the way you suggested, or
- Simply add the credentials manually without relying on Guice, or
- Just do what you suggested, it isn't a big change and if we want to separate this later we can always do it

Wdyt?

**For (b) and (c):**
I think this is much nicer, while it also ties `ContainerS3Facade` to our `ManagedS3MockContainer`, the two are essentially made to go hand-in-hand. We wouldn't expose `ContainerS3Facade` under a `proxy-testing` package - or if we do, it's at least separate from `TestingRemoteS3Facade`

Reply:
For **a)** - I'd do either _ add the credentials manually_ or _leave what I have_. wdyt?

Reply:
I went with the second option for (a), simply adding the credentials by hand in the same way the TestingServer does upon startup - lmk your thoughts!